tests/commandline/test_complex_numbers.csd

```diff
@@ -0,0 +1,22 @@
+<CsoundSynthesizer>
+<CsOptions>
+-n
+</CsOptions>
+<CsInstruments>
+
+instr 1
+Ca:Complex[] init 10
+ca:Complex = 2,1
+print real(ca)
+cb:Complex = polar(ca)
+print arg(cb)
+Ca[0] = ca * ca
+cc:Complex = Ca[0]
+print abs(cc)
+endin
```

Comment:
General note: our tests generally check that they compile correctly but not so much that opcodes run as expected. I added a couple of assert UDOs here: https://github.com/csound/csound/blob/develop/tests/commandline/udo/pass_by_ref.csd#L12-L24

that allow for testing values and failing if not true. I think we could benefit from extracting that out to a reusable library and start to add use of it in our test suite.

Reply:
right, I hadn't come across exitnow(), very useful for this.